### DISCUSSION TITLE
[Beta] A Zephr Heights Mystery & MelonLoader

### DISCUSSION THREAD

**Discussion by wd357dui** [wd357dui/Elements-of-Harmony#8] [Announcements]
# First update following AZHM's release 🎉

I haven't tested it much so it might be buggy 😅

- [x] ~~ACES is tested for AZHM~~ (*after combining test results with DJTHED's I found out that the base game has no ACES, turning it on can only result in turning all tonemapping off!*), but I don't know if I set the exposure settings right, change the setting `DirectXHook.URP.ColorAdjustments.PostExposure` in `Elements of Harmony/Settings.txt` to find the best value
- [x] Fixed the Chinese language not displaying bug caused by the game's incorrect language mappings
- [ ] Custom localization is not yet featured for AZHM

## Install for [MelonLoader](https://github.com/LavaGang/MelonLoader)

Unzip `publish.zip`, and copy contents in the `MelonLoader` folder to the game folder, same goes for both games

## Install standalone

Unzip `publish.zip`, and copy contents in the `Standalone` folder to the game folder, same goes for both games

<hr /><em>This discussion was created from the release <a href='https://github.com/wd357dui/Elements-of-Harmony/releases/tag/v0.3.0-beta'>[Beta] A Zephr Heights Mystery & MelonLoader</a>.</em>

**Comment by Glimka** [comment]
What about correcting existing localizations? Is it possible or not yet?
I could improve Polish because it has some issues but I would need all text in an easily editable file.

**Reply by wd357dui** [reply]
yes it's definitely possible 🥰, I already saw the codes for matching translations, it's not so different from AMBA, basically the same framework, I just need to find where the original translation files are

**Reply by wd357dui** [reply]
ok, though I wasn't able to find them in the game files, I managed to get them at runtime through code, here's the term mappings for Polish, in tab-separated values format:
[pl.txt](https://github.com/wd357dui/Elements-of-Harmony/files/15382814/pl.txt)
and here's the English version for reference:
[en-US.txt](https://github.com/wd357dui/Elements-of-Harmony/files/15382825/en-US.txt)

I'll now work on the code for enabling custom localizations and other bug fixes, will release in a few days 😉

**Reply by Glimka** [reply]
Thanks. I compared both and I see some of the texts have missing translation as in they are empty. I don't know if they forgot to translate them or they are not used in game so they had no reason to do so.
I'll check later (if I'll have a chance) if too long translation is causing (at least) two cutscenes to break and if I can fix that by making translation shorter.

Also speaking of custom localizations I think it would be a good idea to add option to change voice acting language so people who would like to play for example with original cast but have text in game translated could select it in settings.
The easiest solution would be to add existing languages which have dubbed version as additional ones with original voice acting.

**Reply by wd357dui** [reply]
yes, it is a good idea, it is common for people in my country to watch movies / play games with *original English audio + translated subtitles* as well (it just hasn't come up yet because we didn't have Chinese dub for the game to begin with anyway), this feature itself is not hard to implement, but it is hard for me to add new menu options in the game (or any kind of UI elements) since I'm not that familiar with Unity, so the option to use a different voice language would be to change this in `Settings.txt` manually.

**Reply by wd357dui** [reply]
@Glimka check out the new release 🥰 #9
